server/container/util/src/main/java/org/apache/james/util/ReactorUtils.java

```diff
@@ -37,45 +37,35 @@
 import reactor.core.publisher.Mono;
 import reactor.core.publisher.Signal;
 import reactor.core.publisher.SynchronousSink;
+import reactor.core.scheduler.Schedulers;
 import reactor.util.context.Context;
 import reactor.util.function.Tuple2;
 
 public class ReactorUtils {
 
     public static final String MDC_KEY_PREFIX = "MDC-";
 
-    public static class Throttler<T, U> {
-        private static final Duration DELAY = Duration.ZERO;
-
-        public static <T, U> RequiresWindowingParameters<T, U> forOperation(Function<T, Publisher<U>> operation) {
-            return (maxSize, duration) -> new Throttler<>(operation, maxSize, duration);
-        }
-
-        @FunctionalInterface
-        public interface RequiresWindowingParameters<T, U> {
-            Throttler<T, U> window(int maxSize, Duration duration);
-        }
-
-        private Throttler(Function<T, Publisher<U>> operation, int windowMaxSize, Duration windowDuration) {
-            Preconditions.checkArgument(windowMaxSize > 0, "'windowMaxSize' must be strictly positive");
-            Preconditions.checkArgument(!windowDuration.isNegative(), "'windowDuration' must be strictly positive");
-            Preconditions.checkArgument(!windowDuration.isZero(), "'windowDuration' must be strictly positive");
-
-            this.operation = operation;
-            this.windowMaxSize = windowMaxSize;
-            this.windowDuration = windowDuration;
-        }
-
-        private final Function<T, Publisher<U>> operation;
-        private final int windowMaxSize;
-        private final Duration windowDuration;
+    private static final Duration DELAY = Duration.ZERO;
+
+    public static <T> RequiresQuantity<T> throttle() {
+        return elements -> duration -> {
+            Preconditions.checkArgument(elements > 0, "'windowMaxSize' must be strictly positive");
+            Preconditions.checkArgument(!duration.isNegative(), "'windowDuration' must be strictly positive");
+            Preconditions.checkArgument(!duration.isZero(), "'windowDuration' must be strictly positive");
+            return flux -> flux
+                .windowTimeout(elements, duration)
+                .zipWith(Flux.interval(DELAY, duration), 1)
```

Comment:
Could you make more clear of `1`, days or minus or second?

Reply:
concurrency. if we can make it work, we'll refine the PR